### PR TITLE
core(link-elements): gracefully handle header parser error

### PR DESCRIPTION
**Pull request by adamraine**
Part of https://github.com/GoogleChrome/lighthouse/issues/14934

Screenshot 2023-03-28 at 12 16 28 PM

### CODE REVIEW

**Review by paulirish — APPROVED** [review]

File: core/gather/gatherers/link-elements.js [hunk 1]
```diff
@@ -126,18 +139,29 @@ class LinkElements extends FRGatherer {
...
-          rel: link.rel || '',
-          href: normalizeUrlOrNull(link.uri, context.baseArtifacts.URL.finalDisplayedUrl),
-          hrefRaw: link.uri || '',
-          hreflang: link.hreflang || '',
-          as: link.as || '',
-          crossOrigin: getCrossoriginFromHeader(link.crossorigin),
-          source: 'headers',
-          fetchPriority: link.fetchpriority,
-          node: null,
+      try {
```
Comment (brendankenny) [inline core/gather/gatherers/link-elements.js:142]: nit: just try/catch the `LinkHeader.parse()` call? That's the only place we expect an error of this kind, so keeps handling nice and compact?

File: core/gather/gatherers/link-elements.js [hunk 2]
```diff
@@ -19,6 +21,17 @@ import {MainResource} from '../../computed/main-resource.js';
  * headers of the main resource.
  */
 
+const UIStrings = {
+  /**
+   * @description Warning message explaining that there was an error parsing a link header in an HTTP response. `error` will be an english string with more details on the error. `header` will be the value of the header that caused the error. `link` is a type of HTTP header and should not be translated.
+   * @example {Expected attribute delimiter at offset 94} error
+   * @example {<https://assets.calendly.com/assets/booking/css/booking-d0ac32b1.css>; rel=preload; as=style; nopush} error
+   */
+  headerParseWarning: 'Error parsing `link` header ({error}):\n`{header}`',
```
Comment (brendankenny) [inline core/gather/gatherers/link-elements.js:30]: doesn't this `\n` just turn into a space when inserted into the report?

**Review by brendankenny — COMMENTED** [review]
the try/catch is great in general (even a really bad link header shouldn't break _all_ link headers) and LGTM, but for the specific case in #14934, are we sure that is an error? It feels weird to put a top level warning on that case if it isn't

**Comment by adamraine** [comment]
> but for the specific case in https://github.com/GoogleChrome/lighthouse/issues/14934, are we sure that is an error? It feels weird to put a top level warning on that case if it isn't

No this PR does not close #14934, we should still file upstream about that.

**Pull request merged**